src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -416,35 +425,34 @@ private PaymentTransactionInfoPlugin executeFollowUpTransaction(final Transactio
         final DateTime utcNow = clock.getUTCNow();
 
         final PaymentModificationResponse response;
-        try {
-            response = transactionExecutor.execute(transactionAmount, paymentProvider, pspReference, splitSettlementData);
-        } catch (final ModificationFailedException e) {
-            return new AdyenPaymentTransactionInfoPlugin(kbPaymentId, kbTransactionId, transactionType, amount, currency, PaymentServiceProviderResult.ERROR, utcNow, e);
+        response = transactionExecutor.execute(transactionAmount, paymentProvider, pspReference, splitSettlementData);
+        if (!response.isSuccess()) {
+            return new AdyenPaymentTransactionInfoPlugin(kbPaymentId, kbTransactionId, transactionType, amount, currency, Optional.<PaymentServiceProviderResult>absent(), utcNow, response);
```

Comment:
> unclear though if it impacts the logic when we read from the table

Writing non-successful responses to the table might break here: [PluginPaymentDao.java#L143](https://github.com/killbill/killbill-plugin-framework-java/blob/master/src/main/java/org/killbill/billing/plugin/dao/payment/PluginPaymentDao.java#L143)
```java
// Assumes that the last auth was successful
public RESP_R getSuccessfulAuthorizationResponse(final UUID kbPaymentId, final UUID kbTenantId) throws SQLException {
    ...
}
```